cmd/src/actions_exec_backend_runner.go

```diff
@@ -363,3 +359,62 @@ func repositoryZipArchiveURL(repoName, rev, token string) (*url.URL, error) {
 	u.Path = path.Join(u.Path, repoName+"@"+rev, "-", "raw")
 	return u, nil
 }
+
+func unzip(zipFile, dest string) error {
+	r, err := zip.OpenReader(zipFile)
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+
+	for _, f := range r.File {
+		fpath := filepath.Join(dest, f.Name)
+
+		// Check for ZipSlip. More Info: http://bit.ly/2MsjAWE
+		if !strings.HasPrefix(fpath, filepath.Clean(dest)+string(os.PathSeparator)) {
+			return fmt.Errorf("%s: illegal file path", fpath)
+		}
+
+		if f.FileInfo().IsDir() {
+			err := os.MkdirAll(fpath, os.ModePerm)
```

Comment:
This is of course fine as you wrote it, but I'd just one-line it as elsewhere:

```suggestion
			if err := os.MkdirAll(fpath, os.ModePerm); err != nil {
```